sdk/core/Azure.Core.Experimental/src/DynamicRequest.cs

```diff
@@ -0,0 +1,131 @@
+                {
+                    Body = new DynamicJson(sr.ReadToEnd());
+                }
+                Request.Content = value;
+            }
+        }
+
+        // TODO(matell): How does the initialization here play into the ability to send a request with an "empty" body?
+        /// <summary>
+        /// The JSON body of request.
+        /// </summary>
+        public DynamicJson Body { get; set; } = DynamicJson.Object();
+
+        // TODO(matell): In Krzysztof's prototype we also took DiagnosticScope as a parameter, do we still need that?
+        /// <summary>
+        /// Creates an instance of <see cref="RequestContent"/> that wraps <see cref="DynamicJson"/> content.
+        /// </summary>
+        /// <param name="request">The <see cref="Request"/> to send.</param>
+        /// <param name="pipeline">The HTTP pipeline for sending and receiving REST requests and responses.</param>
+        public DynamicRequest(Request request, HttpPipeline pipeline)
+        {
+            Request = request;
+            HttpPipeline = pipeline;
+        }
+
+        /// <summary>
+        /// Send the request asynchronously.
+        /// </summary>
+        /// <param name="cancellationToken">The cancellation token.</param>
+        /// <returns>The response dynamically typed in a <see cref="DynamicResponse"/>.</returns>
+        public async Task<DynamicResponse> SendAsync(CancellationToken cancellationToken = default)
+        {
+            // Since we are sending the underlying request, we need to copy the Content on to it, or we'll lose the body.
+            Request.Content = Content;
+
+            Response res = await HttpPipeline.SendRequestAsync(Request, cancellationToken).ConfigureAwait(false);
+            DynamicJson dynamicContent;
+
+            if (res.ContentStream != null)
+            {
+                using (StreamReader sr = new StreamReader(res.ContentStream, encoding: Utf8NoBom, detectEncodingFromByteOrderMarks:true, bufferSize: 1024, leaveOpen: true))
+                {
+                    dynamicContent = new DynamicJson(await sr.ReadToEndAsync().ConfigureAwait(false));
+                }
+            }
+            else
+            {
+                //TODO(chamons) - Is this correct?
+                dynamicContent = new DynamicJson("");
+            }
+
+            return new DynamicResponse(res, dynamicContent);
+        }
+
+        /// <summary>
+        /// Send the request synchronously.
+        /// </summary>
+        /// <param name="cancellationToken">The cancellation token.</param>
+        /// <returns>The response dynamically typed in a <see cref="DynamicResponse"/>.</returns>
+#pragma warning disable AZC0107 // DO NOT call public asynchronous method in synchronous scope.
+        //TODO(chamons) - Is this correct?
+        public DynamicResponse Send(CancellationToken cancellationToken = default) => SendAsync(cancellationToken).EnsureCompleted();
```

Comment:
We should call `HttpPipeline.SendReques` in the sync overload.

Reply:
Yep, but I want to get the null response/empty string response pinned down first.